File: app/src/main/java/com/thanksmister/bitcoin/localtrader/ui/adapters/PredictAdapter.java
/*
 * Copyright (c) 2018 ThanksMister LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed
 * under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.thanksmister.bitcoin.localtrader.ui.adapters;

import android.content.Context;
import android.location.Address;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.Filterable;
import android.widget.TextView;

import com.thanksmister.bitcoin.localtrader.R;

import java.util.Collections;
import java.util.List;

public class PredictAdapter extends ArrayAdapter<Address> implements Filterable {
    private List<Address> data = Collections.emptyList();
    private final LayoutInflater inflater;

    public PredictAdapter(Context context, List<Address> data) {
        super(context, 0, data);
        this.inflater = LayoutInflater.from(context);
        this.data = data;
    }

    @Override
    public boolean isEnabled(int position) {
        return true;
    }

    @Override
    public int getCount() {
        return data.size();
    }

    @Override
    public Address getItem(int position) {
        return data.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    public void replaceWith(List<Address> data) {
        this.data = data;
        notifyDataSetChanged();
    }

    @Override
    public View getView(int position, View view, ViewGroup parent) {
        ViewHolder holder;
        if (view != null) {
            holder = (ViewHolder) view.getTag();
        } else {
            view = inflater.inflate(R.layout.adapter_address, parent, false);
            holder = new ViewHolder(view);
            view.setTag(holder);
        }

        Address address = getItem(position);
        String addressLine = address.getMaxAddressLineIndex() > 0 ? address.getAddressLine(0) : null;
        String output = "";
        if (addressLine != null) output += addressLine;
        if (address.getLocality() != null) output += ", " + address.getLocality();
        if (address.getCountryName() != null) output += ", " + address.getCountryName();

        holder.addressText.setText(output);

        return view;
    }

    static class ViewHolder {
        TextView addressText;
        public ViewHolder(View view) {

        }
    }
}

File: app/src/main/java/com/thanksmister/bitcoin/localtrader/ui/adapters/PredictAdapter.kt
/*
 * Copyright (c) 2020 ThanksMister LLC
 *  http://www.thanksmister.com
 *
 *  Mozilla Public License 2.0
 *
 *  Permissions of this weak copyleft license are conditioned on making
 *  available source code of licensed files and modifications of those files
 *  under the same license (or in certain cases, one of the GNU licenses).
 *  Copyright and license notices must be preserved. Contributors provide
 *  an express grant of patent rights. However, a larger work using the
 *  licensed work may be distributed under different terms and without source
 *  code for files added in the larger work.
 */

package com.thanksmister.bitcoin.localtrader.ui.adapters

import android.content.Context
import android.location.Address
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.Filterable
import android.widget.TextView

import com.thanksmister.bitcoin.localtrader.R

class PredictAdapter(context: Context, data: List<Address>) : ArrayAdapter<Address>(context, 0, data), Filterable {
    private var data = emptyList<Address>()
    private val inflater: LayoutInflater = LayoutInflater.from(context)

    init {
        this.data = data
    }

    override fun isEnabled(position: Int): Boolean {
        return true
    }

    override fun getCount(): Int {
        return data.size
    }

    override fun getItem(position: Int): Address? {
        return data[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    fun replaceWith(data: List<Address>) {
        this.data = data
        notifyDataSetChanged()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view: View?
        val holder: ViewHolder
        if (convertView != null) {
            view = convertView
            holder = view.tag as ViewHolder
        } else {
            view = inflater.inflate(R.layout.adapter_address, parent, false)
            holder = ViewHolder(view!!)
            view.tag = holder
        }
        val address = getItem(position)
        val addressLine = if (address!!.maxAddressLineIndex > 0) address.getAddressLine(0) else null
        var output = ""
        if (addressLine != null) output += addressLine
        if (address.locality != null) output += ", " + address.locality
        if (address.countryName != null) output += ", " + address.countryName
        holder.addressText!!.text = output
        return view
    }

    internal class ViewHolder(view: View) {
        var addressText: TextView? = null

        init {
            addressText = view.findViewById<TextView>(R.id.addressText)
        }
    }
}